chore: Added HW plugin properties to supported properties in AUTO_BATCH plugin

adisan-dev opened the pull request:
### Details:
- Added the properties from `m_compiled_model_without_batch` to the supported properties.
- returns the merged list of properties.

### Tickets:
 Addresses this issue - https://github.com/openvinotoolkit/openvino/issues/23074

### Testing

Running this command - `benchmark_app -d GPU  -hint throughput -m  <batchable model, mobilenet-v2 for example>`

BEFORE:
```
[Step 8/11] Querying optimal runtime parameters
[ INFO ] Model:
[ INFO ]   OPTIMAL_NUMBER_OF_INFER_REQUESTS: 16
[ INFO ]   NETWORK_NAME: main_graph
[ INFO ]   EXECUTION_DEVICES: GPU.0
[ INFO ]   AUTO_BATCH_TIMEOUT: 1000
```

AFTER:

```
[Step 8/11] Querying optimal runtime parameters
[ INFO ] Model:
[ INFO ]   AUTO_BATCH_TIMEOUT: 1000
[ INFO ]   NETWORK_NAME: main_graph
[ INFO ]   OPTIMAL_NUMBER_OF_INFER_REQUESTS: 16
[ INFO ]   PERF_COUNT: NO
[ INFO ]   ENABLE_CPU_PINNING: NO
[ INFO ]   ENABLE_CPU_RESERVATION: NO
[ INFO ]   MODEL_PRIORITY: MEDIUM
[ INFO ]   GPU_HOST_TASK_PRIORITY: MEDIUM
[ INFO ]   GPU_QUEUE_PRIORITY: MEDIUM
[ INFO ]   GPU_QUEUE_THROTTLE: MEDIUM
[ INFO ]   GPU_ENABLE_LOOP_UNROLLING: YES
[ INFO ]   GPU_DISABLE_WINOGRAD_CONVOLUTION: NO
[ INFO ]   CACHE_DIR: 
[ INFO ]   CACHE_MODE: optimize_speed
[ INFO ]   PERFORMANCE_HINT: THROUGHPUT
[ INFO ]   EXECUTION_MODE_HINT: PERFORMANCE
[ INFO ]   COMPILATION_NUM_THREADS: 12
[ INFO ]   NUM_STREAMS: 2
[ INFO ]   PERFORMANCE_HINT_NUM_REQUESTS: 0
[ INFO ]   INFERENCE_PRECISION_HINT: f16
[ INFO ]   DYNAMIC_QUANTIZATION_GROUP_SIZE: 32
[ INFO ]   ACTIVATIONS_SCALE_FACTOR: -1
[ INFO ]   KV_CACHE_PRECISION: i8
[ INFO ]   DEVICE_ID: 0
[ INFO ]   EXECUTION_DEVICES: GPU.0
```

p-wysocki commented:
Thanks for the PR!

bump @zhaixuejun1993

zhaixuejun1993 commented:
Can we avoid to duplicated this two info?
[ INFO ]   NETWORK_NAME: main_graph
[ INFO ]   EXECUTION_DEVICES: GPU.0

adisan-dev commented:
@zhaixuejun1993 

I have removed the dupllicated properties (there were three of them) and updated PR desc:
```
[ INFO ]   OPTIMAL_NUMBER_OF_INFER_REQUESTS: 16
[ INFO ]   NETWORK_NAME: main_graph
[ INFO ]   EXECUTION_DEVICES: GPU.0
```

PTAL.
Another approach could be inserting all properties in a set data structure to get rid of the duplicate ones. But it will add additional complexity.

zhaixuejun1993 reviewed (approved):
LGTM

mlukasze commented:
build_jenkins

mlukasze commented:
ping @zhaixuejun1993

peterchen-intel commented:
@adisan-dev Can you please fix the code style issues?
